### DISCUSSION TITLE
How to write a pointpillars detection demo in cpp

### DISCUSSION THREAD

**Discussion by Guoxs** [open-mmlab/mmdeploy#1706] [Q&A]
I have generated pointpillars' engine file and the performance looks good, now I want to write a 3D object detection inference demo using C++. However,  I can only find c++ inference demo for 2D object detection (demo/csrc/cpp/detector.cxx):

![image](https://user-images.githubusercontent.com/11255550/216305764-90e580b9-6d82-42c4-b3db-656315933a4e.png)

It seems that the **detector.Apply** API can only support Mat object.  So I want to know how to make this API support pointcloud type, such as pcl::PointCloud<pcl::PointXYZI>. 

Thanks!

**Comment by lvhan028** [comment]
@lzhangzz
